--- a/platform/macos/macos.cmake
+++ b/platform/macos/macos.cmake
@@ -80,6 +80,7 @@
     )
 endif()
 
+if(BUILD_TESTING)
 add_executable(
     mbgl-test-runner
     ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/test/main.cpp
@@ -130,6 +131,7 @@
 set_property(TARGET mbgl-test-runner PROPERTY FOLDER Executables)
 set_property(TARGET mbgl-render-test-runner PROPERTY FOLDER Executables)
 set_target_properties(mbgl-benchmark-runner mbgl-test-runner mbgl-render-test-runner PROPERTIES XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH[variant=Debug] "YES")
+endif()
 
 # Disable benchmarks in CI as they run in VM environment
 if(NOT DEFINED ENV{CI})